core/client/fs/src/main/java/alluxio/client/file/BaseFileSystem.java

```diff
@@ -172,16 +173,11 @@ private boolean checkDirectAccess(AlluxioURI uri) {
     if (!getConf().isSet(PropertyKey.USER_FILE_DIRECT_ACCESS)) {
       return false;
     }
-    if (mPathList == null) {
-      mPathList = getConf().getList(PropertyKey.USER_FILE_DIRECT_ACCESS);
+    if (mPathRegex == null) {
+      mPathRegex = Pattern.compile(
```

Comment:
Wondering if there is concurrency issue?

Reply:
i thought about using AtomicReference too, but reference updates are always atomic anyways, so in the worst case, we end up compiling a number of times more than necessary? but they should result in the same value?